gravitee-am-service/src/main/java/io/gravitee/am/service/model/PatchFactorSettings.java

```diff
@@ -36,14 +36,14 @@ public class PatchFactorSettings {
 
     public FactorSettings patch(FactorSettings _toPatch) {
         FactorSettings toPatch = _toPatch == null ? new FactorSettings() : new FactorSettings(_toPatch);
-        if ((hasText(toPatch.getDefaultFactorId()) && defaultFactorId.isEmpty())
-                || (toPatch.getApplicationFactors() == null || toPatch.getApplicationFactors().isEmpty()) && (applicationFactors.isEmpty() || applicationFactors.get().isEmpty())) {
-            throw new InvalidParameterException("Default factor is not set");
-        }
         SetterUtils.safeSet(toPatch::setDefaultFactorId, getDefaultFactorId());
         getApplicationFactors().ifPresent(patchApplicationFactorSettings -> toPatch.setApplicationFactors(
                 applicationFactorsPath(toPatch.getApplicationFactors(), patchApplicationFactorSettings)
         ));
+        boolean noFactors = toPatch.getApplicationFactors() == null || toPatch.getApplicationFactors().isEmpty();
+        if(!noFactors && !hasText(toPatch.getDefaultFactorId())){
```

Comment:
suggestion: can we have a 'hasFactors' boolean instead of 'noFactors', in order to avoid a ("not" noFactors) ?  this is not a blocker but I think is will be more readable. wdyt ?

Reply:
Of course, that's a valid point.